lib/cfn-model/transforms/serverless.rb

```diff
@@ -0,0 +1,82 @@
+class CfnModel
+  class Transforms
+    # Handle transformation of model elements performed by the
+    # Serverless trasnform, see
+    # https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/transform-aws-serverless.html
+    class Serverless
+      def perform_transform(cfn_hash)
+        resources = cfn_hash['Resources'].clone
+        resources.each do |r_name, r_value|
+          next unless r_value['Type'].eql? 'AWS::Serverless::Function'
+          replace_serverless_function cfn_hash, r_name
+        end
+      end
+
+      def self.instance
+        @instance ||= Serverless.new
+        @instance
+      end
+
+      private
+
+      def replace_serverless_function(cfn_hash, resource_name)
+        resource = cfn_hash['Resources'][resource_name]
+        # Bucket is 3rd element of an S3 URI split on '/'
+        code_bucket = \
```

Comment:
the \ drives my eyes nuts in RubyMine, but that aside - suggest a named method to encapsulate the extraction of the token/element